safe/gui/tools/peta_jakarta_dialog.py

```diff
@@ -537,3 +528,32 @@ def reject(self):
         # add our own logic here...
 
         super(PetaJakartaDialog, self).reject()
+
+    def download(self, url, output_path):
+        """Download file from API url and write to output path.
+
+        :param url: URL of the API.
+        :type url: str
+
+        :param output_path: Path of output file,
+        :type output_path: str
+        """
+        request_failed_message = self.tr(
+            "Can't access PetaBencana API: {source}").format(
+            source=url)
+        downloader = FileDownloader(url, output_path)
+        result = downloader.download()
+        if not result:
+            display_warning_message_box(
+                self,
+                self.tr('Download error'),
+                self.tr(request_failed_message))
+            raise Exception(request_failed_message)
```

Comment:
No, don't raise an exception. We never want to display an exception to the user. We want to avoid this as much as possible.